src/Chapter4.hs

```diff
@@ -628,8 +666,13 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
+
+-- practice task to better understand binding
+notM :: (Monad m) => m Bool -> m Bool
+notM x = x >>= (\a -> pure (not a))
+
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM a b = a >>= (\x -> if x then b >>= (\y -> pure $ x && y) else pure False)
```

Comment:
Your implementation is correct :+1: The main idea is to use `if-then-else` inside lambda to achieve the desired lazy behaviour. Though, your implementation can be simplified a bit :slightly_smiling_face: 

```suggestion
andM a b = a >>= \x -> if x then b else pure False
```

Key observation here is that `True && y` is always equal to `y`, so no need to apply `&&` to `x`